openassessment/fileupload/api.py

```diff
@@ -174,15 +219,156 @@ class FileUploadManager(object):
     def __init__(self, openassessment_xblock):
         self.block = openassessment_xblock
 
+    @cached_property
+    def student_item_dict(self):
+        """ Returns a dict containing 'student_id', 'course_id', and 'item_id'. """
+        return self.block.get_student_item_dict()
+
+    @property
+    def team_id(self):
+        """ Returns the current team_id or None. """
+        return self.block.team.team_id if self.block.has_team() else None
+
     def get_uploads(self, include_deleted=False):
         """
         Returns:
-            A list of FileUpload objects associated with an instance of an O.A. Block.
+            A list of FileUpload objects associated with an instance of an Open Assessment Block.
+            This will include FileUpload objects corresponding to existing files.
+            It does **not** include entries for files that have been deleted.
         """
         descriptions, names, sizes = self._get_metadata_from_block()
-        return self._file_uploads_from_list_fields(
-            descriptions, names, sizes, include_deleted
+        user_uploads = self._file_uploads_from_list_fields(descriptions, names, sizes, include_deleted=include_deleted)
+
+        if self.block.is_team_assignment():
+            return self._uploads_owned_by_current_user_and_team(user_uploads)
+
+        return user_uploads
+
+    def get_team_uploads(self):
+        if self.block.is_team_assignment():
+            return self._uploads_owned_by_teammates()
+        return []
+
+    def _uploads_owned_by_current_user_and_team(self, user_uploads):
```

Comment:
Maybe rename this function? Based on the name I would assume this would also return teammates shared uploads

Reply:
👍